deeprank/generate/DataGenerator.py

```diff
@@ -33,32 +35,23 @@ def tqdm(x):
 def _printif(string, cond): return print(string) if cond else None
 
 
+_log = logging.getLogger(__name__)
+
+
 class DataGenerator(object):
 
-    def __init__(self, chain1, chain2,
-                 pdb_select=None, pdb_source=None,
-                 pdb_native=None, pssm_source=None, align=None,
+    def __init__(self, mutants,
                  compute_targets=None, compute_features=None,
                  data_augmentation=None, hdf5='database.h5',
                  mpi_comm=None):
         """Generate the data (features/targets/maps) required for deeprank.
 
         Args:
-            chain1 (str): First chain ID
-            chain2 (str): Second chain ID
-            pdb_select (list(str), optional): List of individual conformation for mapping
-            pdb_source (list(str), optional): List of folders where to find the pdbs for mapping
-            pdb_native (list(str), optional): List of folders where to find the native comformations,
-                nust set it if having targets to compute in parameter "compute_targets".
-            pssm_source (list(str), optional): List of folders where to find the PSSM files
-            align (dict, optional): Dicitionary to align the compexes,
-                                    e.g. align = {"selection":{"chainID":["A","B"]},"axis":"z"}}
-                                    e.g. align = {"selection":"interface","plane":"xy"}
-                                    if "selection" is not specified the entire complex is used for alignement
+            mutants (list(PdbMutantSelection)): the selected mutants
             compute_targets (list(str), optional): List of python files computing the targets,
                 "pdb_native" must be set if having targets to compute.
             compute_features (list(str), optional): List of python files computing the features
-            data_augmentation (int, optional): Number of rotation performed one each complex
+            data_augmentation (int, optional): Number of rotations performed one each pdb
             hdf5 (str, optional): name of the hdf5 file where the data is saved, default to 'database.h5'
```

Comment:
What is data augmentation used for? Which step?

Reply:
This setting seems to trigger the use of the "_add_aug_pdb" function within the DataGenerator class. This function applies the transformation, provided in the "align" setting a given number of times. In the example however, it's set to 0, meaning no augmentation steps. We disabled the align setting, so this setting will always do random rotations around the mutation center to create multiple versions of the same grid feature.

Reply:
I noticed its relation with "align"
So are the multiple versions redundant?

Reply:
We should ask the authors to be sure. I can imagine that these multiple versions can assist in training the neural network.

Reply:
See Li's comment in slack. Deeprank exposes the network to several rotations to improve results.

Reply:
It seems the axis/plant option isn't really used for the rotation. So maybe we should just leave the data augmentation and align argument in the code.

Reply:
Thanks Coos, yes we shall leave the data augmentation and the align arg.

Reply:
OK, give me some time to put the align option back in the class..

Reply:
Sure, please take your time.
I am sorry about the confusion

Reply:
The align option is now back. Unfortunately I cannot make the mutant-based data generator class support interface alignment, since the input is not a protein-protein interface.

Reply:
This is fine. We don't require it to support interface alignment